src/Cake.Npm/Install/NpmInstallSettingsExtensions.cs

```diff
@@ -106,6 +106,53 @@ public static NpmInstallSettings ForProduction(this NpmInstallSettings settings)
         }
 
         /// <summary>
+        /// Defines that npm should npm should prevent optional dependencies from being installed
+        /// </summary>
+        /// <param name="settings">The settings.</param>
+        /// <returns>The <paramref name="settings"/> instance with <see cref="NpmInstallSettings.NoOptional"/> set to <c>true</c>.</returns>
+        public static NpmInstallSettings WithNoOptional(this NpmInstallSettings settings)
```

Comment:
IMHO `WithouthOptionalDependencies` would be clearer to understand the intention of the method.